pkg/services/dashboards/dashboard_service.go

```diff
@@ -111,6 +111,11 @@ func (dr *dashboardServiceImpl) buildSaveDashboardCommand(dto *SaveDashboardDTO,
 		return nil, models.ErrDashboardUpdateAccessDenied
 	}
 
+	err := dr.validateDashboardIsNotProvisioned(dash.Id)
```

Comment:
This should probably be done before ValidateDashboardBeforeSaveCommand since the dashboardId can be changed in that validation if title changed for example.